docs/logs/extending-the-sdk/MyClassWithRedactionEnumerator.cs

```diff
@@ -0,0 +1,72 @@
+// <copyright file="MyRedactionProcessor.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+
+internal class MyClassWithRedactionEnumerator : IReadOnlyList<KeyValuePair<string, object>>
+{
+    private static readonly Regex Rgx = new("\\(?\\d{3}[\\.|\\)]?\\d{3}[\\.|\\-]\\d{4}");
+    private readonly IReadOnlyList<KeyValuePair<string, object>> myList;
+
+    public MyClassWithRedactionEnumerator(IReadOnlyList<KeyValuePair<string, object>> traits)
+    {
+        this.myList = traits;
+    }
+
+    public int Count => this.myList.Count;
+
+    public KeyValuePair<string, object> this[int index] => this.myList[index];
+
+    public IEnumerator<KeyValuePair<string, object>> GetEnumerator()
+    {
+        foreach (var entry in this.myList)
+        {
+            var entryVal = entry.Value;
+            if (entryVal != null && Rgx.IsMatch(entryVal.ToString()))
+            {
+                yield return new KeyValuePair<string, object>(entry.Key, "newRedactedValueHere");
+            }
+            else
+            {
+                yield return entry;
+            }
+        }
+    }
+
+    IEnumerator IEnumerable.GetEnumerator()
+    {
+        return this.GetEnumerator();
+    }
+
+    public override string ToString()
```

Comment:
I feel that this part is not needed and might actually be misleading.

I guess @cijothomas' change would actually help to eliminate the need https://github.com/open-telemetry/opentelemetry-dotnet/pull/3213.

Reply:
We'll need to do serialization.
Alternatives I have in mind:
1. Update MyExporter, which is currently the Exporter used in extending-the-sdk to contain the logic as is in https://github.com/open-telemetry/opentelemetry-dotnet/pull/3213/files, which would add some unnecessary complexity to MyExporter.
2. Instead of using MyExporter as the Exporter for extending-the-sdk, use `ConsoleLogRecordExporter.cs` which already takes care of serialization. But with this, we probably don't need MyExporter anymore.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/3217#discussion_r855526612

Reply:
> . But with this, we probably don't need MyExporter anymore.

Its there to demonstrate how to write own exporter part.

Reply:
> We'll need to do serialization.

No need. This doc (the parts from this PR) is just how to write a custom processor, which can do redaction of properties.

Reply:
I've removed the ToString() override of [MyClassWithRedactionEnumerator.cs](https://github.com/open-telemetry/opentelemetry-dotnet/pull/3217/files/9b54a621507e30eabb67e05b298a2bf05ec744bb#diff-2cd84068fc8e2ab3d7a411c5a7eb10c71ab09b1f0c9d4aa224c9083738029af1). Thanks!